bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolBridgeHandler.java

```diff
@@ -0,0 +1,343 @@
+                }
+                if (!isConnected) {
+                    logger.debug("Cannot establish connection to {} ({})", ipAddress, unconnectedReason);
+                } else {
+                    unconnectedReason = "";
+                }
+                updateStatus();
+            }
+        }
+    }
+
+    private synchronized void startAutomaticRefresh() {
+        ScheduledFuture<?> job = pollingJob;
+        if (job == null || job.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(this::pollingRunnable, 0, refreshInterval, TimeUnit.SECONDS);
+        }
+    }
+
+    public ThingStatus getStatus() {
+        return getThing().getStatus();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No commands supported - nothing to do
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus();
+        ResolBridgeConfiguration configuration = getConfigAs(ResolBridgeConfiguration.class);
+        ipAddress = configuration.ipAddress;
+        refreshInterval = configuration.refreshInterval;
+        password = configuration.password;
+        startAutomaticRefresh();
+    }
+
+    @Override
+    public void dispose() {
+        ScheduledFuture<?> job = pollingJob;
+        if (job != null) {
+            job.cancel(true);
+            pollingJob = null;
+        }
+        try {
+            Connection connection = tcpConnection;
+            if (connection != null) {
+                connection.disconnect();
+                getThing().getThings().stream().forEach(thing -> {
+                    ThingHandler th = thing.getHandler();
+                    if (th instanceof ResolEmuEMThingHandler) {
+                        ((ResolEmuEMThingHandler) th).stop();
+                    }
+                });
+
+            }
+        } catch (IOException ioe) {
+            // we don't care about exceptions on disconnect in dispose
+        }
+    }
+
+    Locale getLocale() {
+        return locale;
+    }
+
+    /* adapter to react on connection state changes and handle received packets */
+    private class ResolConnectorAdapter extends ConnectionAdapter {
+        @Override
+        public void connectionStateChanged(@Nullable Connection connection) {
+            synchronized (ResolBridgeHandler.this) {
+                if (connection == null) {
+                    isConnected = false;
+                } else {
+                    ConnectionState connState = connection.getConnectionState();
+                    if (ConnectionState.CONNECTED.equals(connState)) {
+                        isConnected = true;
+                    } else if (ConnectionState.DISCONNECTED.equals(connState)
+                            || ConnectionState.INTERRUPTED.equals(connState)) {
+                        isConnected = false;
+                    }
+                    logger.info("Connection state changed to: {}", connState.toString());
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug.